packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js

```diff
@@ -1034,6 +1034,46 @@ module('Search', setupModule, () => {
         assert.strictEqual(this.$input.val(), 'PM', 'revert incorrect changes');
     });
 
+    test('when A is pressed it should toggle PM to AM', function(assert) {
+        this.instance.option({
+            value: new Date('10/10/2012 09:00 PM'),
+            useMaskBehavior: true,
+            displayFormat: 'a'
+        });
+
+        this.keyboard.press('a');
+        assert.strictEqual(this.$input.val(), 'AM');
+    });
+
+    test('when P is pressed it should toggle AM to PM', function(assert) {
+        this.instance.option({
+            value: new Date('10/10/2012 09:00 AM'),
+            useMaskBehavior: true,
+            displayFormat: 'a'
+        });
+
+        this.keyboard.press('p');
+        assert.strictEqual(this.$input.val(), 'PM');
+    });
+
+    [
+        { letter: 'x', expectedValue: 'AM' },
+        { letter: 'y', expectedValue: 'AM' },
+        { letter: 'x', expectedValue: 'PM' },
+        { letter: 'y', expectedValue: 'PM' },
+    ].forEach(({ letter, expectedValue }) => {
+        QUnit.test('when other character is pressed it should not toggle PM to AM', function(assert) {
```

Comment:
i suggest this name - 'AM/PM part should not change when keys other than 'a' or 'p' are pressed.'

Reply:
That's better